fabric8-maven-plugin/src/main/java/io/fabric8/maven/JsonMojo.java

```diff
@@ -1157,7 +1163,7 @@ protected boolean addServiceAccountIfIUsingSecretAnnotations(KubernetesListBuild
             secrets.add(secretRef);
         }
 
-        if (!secrets.isEmpty()) {
+        if (!secrets.isEmpty() || createServiceAccount) {
```

Comment:
we should be using much better more fine grained roles / SAs really across devops / ipaas...